[MRG + 1] Add warning for KFold CV if random_state must be changed but shuffle is

alexsavio opened the pull request:
False

Trying the PR in another issue.

alexsavio committed:
Add warning for KFold CV if random_state must be changed but shuffle is
False

amueller commented:
Thanks, looks good.

agramfort commented on sklearn/utils/validation.py:
```diff
@@ -620,3 +620,20 @@ def check_is_fitted(estimator, attributes, msg=None, all_or_any=all):
+    """
+    if random_state is not None and not shuffle:
+        warnings.warn("`shuffle` should be set to True for the data"
+                      " to be shuffled based on the given `random_state`.")
```
there is no test for this function.

please add one in test_validation.py

also interacts with #4550

glennq committed:
MLP bug fixes with random state and loss, removed decision_function, and added test

Commit:
Fixed issue #5830, capitalization in advanced installation documenation

Commit:
FIX n_iter_without_progress and min_grad_norm in TSNE

Adds tests for n_iter_without_progress and min_grad_norm

willingc committed:
Reorganize README

willingc committed:
Remove write clone

willingc committed:
Update install

willingc committed:
Fix changes re: @amuller

willingc committed:
Update mailing list url

stuppie committed:
use sparse contingency matrix for supervised cluster metrics
Remove max_n_classes option

amueller committed:
clean up deprecation warning stuff in common tests

minor fixes in preprocessing tests

jnothman committed:
[MRG + 1] FIX use high precision cumsum and check it is stable enough (#7331)

* FIX use high precision cumsum and check it is stable enough

amueller committed:
[MRG+1] fix sampling in stratified shuffle split (#6472)

Fix sampling in stratified shuffle split, break tests that test sampling.

amueller committed:
[MRG] add warning when importing old or new pickle. (#7248)

zhengruifeng committed:
[MRG+1] Add hyperlink and example for kappa (#7214)

ogrisel committed:
Add Dirichlet process prior to BayesianGaussianMixture (#7386)

jblackburne committed:
Removed isclose from _binary_clf_curve. (#7353)

Added a bugfix report to whats_new.rst. Modified a unit test to check for the kind of problems caused by isclose.

ogrisel committed:
DOC fix for svd_solver in PCA docstring

MechCoder committed:
[MRG] Fail imputer early when number of features are not the same in fit and transform (#7374)

raghavrv committed:
[MRG+1] ENH/MNT Rename labels --> groups in CV tools (#6660)

jnothman committed:
ENH raise error when scoring=sklearn.metrics.{classification,ranking,...}.*

themrmax committed:
make export_graphviz return string if out_file is None

themrmax committed:
remove unused StringIO.get_value

themrmax committed:
fix lint issues

ogrisel committed:
DOC fix class.rst for Label / Group CV tools

ogrisel committed:
ENH more informative validation error message

ogrisel committed:
Merge pull request #7393 from ogrisel/mlp-multilabel

[MRG] Fix for MLP multilabel predict_proba and removal of decision_function

ogrisel committed:
ENH rename l-bfgs to lbfgs in MLP for the sake of consistency

raghavrv committed:
MNT Remove unused constants from public API(?) (#6517)

ogrisel committed:
ENH rename algorithm to solver in MLP for consistency

amueller committed:
link to scikit-learn tag on stackoverflow, minor formatting.

raghavrv committed:
[MRG+1] TST Stronger test for _check_is_permutation (#7395)

* TST Stronger test for _check_is_permutation

* TST Ensure additional duplicate indices are caught

jnothman committed:
DOC mention true positives etc in confusion matrix docs (#7392)

nelson-liu committed:
[MRG+1] edited the FAQ to fix a typo and include info regarding global random states (#5927)

* edited the FAQ to fix a typo and include info regarding global random states.

* more explicitly delineated scikit-learn's use of numpy global random state and added comments about thread safety.

* amended info about multiprocessing and instead discussed how global random states are prone to unintended change. Also used sphinx formatting on numpy.random.seed.

* fixed line breaks

* link to function

* emphasize random_state param for estimators and cv splitters

* edit line length and add backticks to param name

yl565 committed:
FIX 7155: GridSearchCV predict_proba delegation to SDGClassifier

ogrisel committed:
Merge pull request #7309 from amueller/how_to_ask_questions

[MRG + 1] How to ask questions

jamesroutley committed:
Fixing typo (#7402)

Commit:
Added user guide link in function_tranformer docstring (#5785)

* Added user guide link in function_tranformer docstring

* Added function_transformer tag to preprocessing user guide

themrmax committed:
only return string if out_file is None

themrmax committed:
fix indentation

MechCoder committed:
Merge pull request #6400 from scikit-learn/radneigh-njobs

[MRG + 1] ENH use n_jobs in brute-force radius_neighbors

themrmax committed:
fix lint error

ogrisel committed:
Merge pull request #7396 from ogrisel/mlp-optimizer

[MRG] ENH rename l-bfgs to lbfgs in MLP for the sake of consistency

ogrisel committed:
Merge pull request #7401 from ogrisel/sgdclf-predict_proba-delegation

[MRG] FIX 7155: GridSearchCV predict_proba delegation to SDGClassifier

ogrisel committed:
DOC add what's new entry for #7159

[ci skip]

ogrisel committed:
Merge pull request #7404 from amueller/capitalization_rebase

Fixed issue #5830, capitalization in advanced installation documenation

robinschucker committed:
Fixed doc for issue #7409 (#7410)

amueller committed:
COSMIT don't use docstrings in tests (gp kernel tests) (#7399)

jnothman committed:
Fix #6994: MLP partial_fit with unseen classes (#7358)

jnothman committed:
TST fix generated test names in case where test fails

ogrisel committed:
MAINT master is now 0.19.dev0

amueller committed:
make pickle version test more stable (#7415)

amueller committed:
[MRG] add whatsnew for #7248 (unpickle version warning) (#7400)

jnothman committed:
TST move test naming and use inspect

themrmax committed:
retain default behaviour of out_file=tree.dot

ogrisel committed:
FIX use == to avoid relying on string interning

kmike committed:
DOC fix intercept_ and coef_ docs for LogisticRegression(CV) (#7420)

* coef_ can be of shape (1, n_features) for binary problems;
* intercept can be of shape (1,) for binary problems;
* intercept_ is set to zero, not unavailable when fit_intercept is False;
* raw_coef_ is removed (see GH-3416)

TomDLT committed:
DOC fix various syntax in whats_new (#7421)

GaelVaroquaux committed:
ENH: display the original exception (#5906)

Fixes #5905

ogrisel committed:
FIX missing import in setup.py

Commit:
lda_qda doc: QDA vs Naive Bayes correction (#5997)

* lda_qda doc: QDA vs Naive Bayes correction

Small correction in lda_qda.rst re: comparison of QDA with diagonal covariance matrices and Naive Bayes.

In Naive Bayes, it doesn't make sense to talk about conditional independence between the classes. The Naive Bayes assumptions states that given the label, the input features are conditionally independent. This is reflected in the QDA by assuming diagonal covariance matrices for each class.

* Update lda_qda.rst

hallr committed:
added contributors to whats new for 0.16 and 0.17 (#5840)

* added contributors to whats new for 0.16 and 0.17

* added contributors for 016 and 017 as single line, comma delimited

* added contributors for 016 and 017 as single line, comma delimited

amueller committed:
added contributors for 0.18, minor formatting fixes.

themrmax committed:
add depreciation warning for out_file default

themrmax committed:
use double quotes

kmike committed:
use == instead of 'is' for string comparison (#7422)

nelson-liu committed:
DOC add species distributions dataset to classes.rst (#7430)

themrmax committed:
edit depreciation warning and add versionadded to docstring

themrmax committed:
only show DepreciationWarning if out_file paramater is actually missing

lesteve committed:
Change flake8_diff.sh logic for branch in main repo (#7437)

Only commits that were pushed in the branch are tested. They are
provided by the TRAVIS_COMMIT_RANGE environment variable.

rth committed:
[MRG] Text vectorizers memory usage improvement (#7272)

With thanks to @ephes and @aupiff.

jnothman committed:
Make named_check private

hlin117 committed:
[MRG + 1] DOC Default parameter info for KernelPCA (#6035)

* Added default attribute information and more parameter data to KernelPCA docs

* Pulling back changes from rebase

Titan-C committed:
[MAINT] Upgrade to Sphinx-Gallery 0.1.4 (#7354)

* [MAINT] Upgrade to Sphinx-Gallery 0.1.4

* Don't run flake8 on Sphinx-Gallery

* Sphinx-Gallery loads the stylesheet not the layout

* Delete old css file of the gallery

* Incorporate scikit-learn css to wrap download buttons

* flake8 message that sphinx-gallery is not checked

ogrisel committed:
[MRG+2] Faster isotonic rebased (#7444)

stvn66 committed:
fix typo in DBSCAN explanation

stvn66 committed:
remove unnecessary comma

GaelVaroquaux committed:
Merge pull request #7447 from stvn66/stvn66-dbscan-docs-typo

fix typo in DBSCAN explanation

elena-sharova committed:
Provided a fix for issue #7140 - Minor discrepancy in hashing feature
  transformation example.

aabadie committed:
DOC fix deprecation warning in faces decomposition example

themrmax committed:
remove lambda for flake8 linting

aabadie committed:
[MRG+2] DOC: update persistence documentation with Joblib new behavior (#7434)

aabadie committed:
fixing failing faces decomposition example

aabadie committed:
replace RandomizedPCA by new API in remaining documentation

aabadie committed:
add explicit comment in face decomposition example

Commit:
[MRG] DOC Example: Fix minor standard output formatting (#7456)

NelleV committed:
Merge pull request #7449 from elena-sharova/hashing_feature_transform_example_#7140

  Provided a fix for issue #7140 - Minor discrepancy in hashing feature

lesteve committed:
[MRG + 1] DOC replace RandomizedPCA with PCA and svd_solver='randomized' in documentation (#7450)

rsmith54 committed:
Add OneVs{One,All}Classifier._pairwise: fix for #7306 (#7350)

NelleV committed:
Merge pull request #7412 from jnothman/fix_test_descriptions

[MRG+1] TST fix generated test names

themrmax committed:
add object inheritence to class definition

musically-ut committed:
[MRG] ENH: Allow specifying arbitrary kernel for semi-supervised learning. (#5762)

NelleV committed:
Merge pull request #7389 from jnothman/sniff_bad_scoring2

[MRG + 1] ENH raise error when scoring=sklearn.metrics.{classification,ranking,…}

NelleV committed:
Merge pull request #7390 from themrmax/export-graphviz-to-string

[MRG+1] make export_graphviz return string if out_file is None

Commit:
MAINT remove unneeded import (#7465)

doshyt committed:
[MRG] Add Deprecation Warning to address making "batch" the default learning method in LDA in 0.19 release (#6999)

* update default LDA method to batch

* update tests to match default LDA learning scheme

* DOC trim information on faster learning scheme with LDA

* Add deprecation warning and revert learning_method back to the default

After the discussion, it was decided that a proper deprecation cycle
for the parameter is required. Therefore, in this PR only the warning is
added. The default learning_method will be changed to 'batch' in 0.19.

* Revert back tests to the default LDA

* DOC deprecate in 0.20 instead of 0.19

jnothman committed:
[MRG+1] Fix regression in silhouette_score for clusters of size 1 (#7438)

NelleV committed:
Merge pull request #7368 from amueller/deprecation_warnings_tests

[MRG+1] clean up deprecation warning stuff in common tests

willingc committed:
Add @raghavrv suggested changes

amueller committed:
s/Contributors/Code Contributors/g

NelleV committed:
Merge pull request #7018 from willingc/update-readme

Update readme #7003

Commit:
Addressing issue #7468. (#7472)

Removing entry about "indicator features" in `whats_new.rst`, because it was
causing multiple issues and this resulted in being reverted.

amueller committed:
[MRG] bump master to 0.19-dev (#7413)

* bump master to 0.19-dev

* also bump whatsnew

amueller committed:
add some more entries to the mailmap

jnothman committed:
[MRG+2] FIX adaboost estimators not randomising correctly (#7411)

* FIX adaboost estimators not randomising correctly

(fixes #7408)

FIX ensure nested random_state is set in ensembles

* DOC add what's new

* Only affect *__random_state, not *_random_state for now

* TST More informative assertions for ensemble tests

* More specific testing of different random_states

amueller committed:
more mailmap and whatsnew additions

lesteve committed:
Fix typo in whats_new.rst

lesteve committed:
Avoid creating an ambiguous reference for usptream/master

There was a clash between refs/upstream/master and
refs/remotes/upstream/master. That was problematic when running
flake8_diff.sh locally.

GaelVaroquaux committed:
Merge pull request #7428 from amueller/contributors018

[MRG+1] added contributors for 0.18, minor formatting fixes.

hnykda committed:
Sync docstring and definition default argument in kneighbors (#7476)

kirillbobyrev committed:
Minor k_means_.py cleanup.

Fixed two PEP8 warnings about breaking line before binary operator by simply
putting the binary operator on the line before.

Fixed one PEP8 warning about line being too long (extending 79 symbols limit).

Removed `best_inertia = np.infty` line, because `best_inertia` is initialized
again few lines after and never used in between.

kirillbobyrev committed:
Minor test_k_means.py cleanup.

Removed unused import.

Fixed two lines extending 79 width limit.

kirillbobyrev committed:
Revert PEP8 fixes.

NelleV committed:
Merge pull request #7480 from omtcyfz/minor-k-means-cleanup

Minor k_means_.py cleanup.

mlliou112 committed:
DOC: Added Nested Cross Validation Example (#7111)

fukatani committed:
Fix _get_importances. (#7487)

Not to use hasattr.

raghavrv committed:
[MRG+2] Timing and training score in GridSearchCV (#7325)

* Resolved issue #6894 and #6895:
    Now *SearchCV.results_ includes both timing and training scores.

wrote new test (sklearn/model_selection/test_search.py)
and new doctest (sklearn/model_selection/_search.py)
added a few more lines in the docstring of GridSearchCV and RandomizedSearchCV.
Revised code according to suggestions.
Add a few more lines to test_grid_search_results():
    1. check test_rank_score always >= 1
    2. check all regular scores (test/train_mean/std_score) and timing >= 0
    3. check all regular scores <= 1
Note that timing can be greater than 1 in general, and std of regular scores
always <= 1 because the scores are bounded between 0 and 1.

* ENH/FIX timing and training score.

* ENH separate fit / score times
* Make score_time=0 if errored; Ignore warnings in test
* Cleanup docstrings
* ENH Use helper to store the results
* Move fit time computation to else of try...except...else
* DOC readable sample scores
* COSMIT Add a commnent on why time test is >= 0 instead of > 0
  (Windows time.time precision is not accurate enought to be non-zero
   for trivial fits)

* Convey that times are in seconds

jnothman committed:
DOC insert missing whats_new heading

amueller committed:
fix lbfgs rename (#7503)

amueller committed:
[MRG + 1] More versionadded everywhere! (#7403)

* insert versionadded versionchanged directives in docstrings for 0.18

indicate where exception classes were moved from

* moved versionadded in the proper places

nelson-liu committed:
[MRG+3] Fix min_weight_fraction_leaf to work when sample_weights are not provided (#7301)

* fix min_weight_fraction_leaf when sample_weights is None

* fix flake8 error

* remove added newline and unnecessary assignment

* remove max bc it's implemented in cython and add interaction test

* edit weight calculation formula and add test to check equality

* remove test that sees if two parameter build the same tree

* reword min_weight_fraction_leaf docstring

* clarify uniform weight in forest docstrings

* update docstrings for all classes

* add what's new entry

* move whatsnew entry to bug fixes and explain previous behavior

TomDLT committed:
Fix typo in PCA docstring

GaelVaroquaux committed:
DOC: trivial rst fix to whats_new.rst (#7511)

Fix for proper rendering

jnothman committed:
[MRG] DOC Attempt to impose some order on What's New 0.18 (#7496)

* DOC Attempt to impose some order on What's New 0.18

* DOC more what's new subdivision

* DOC add heading to fix heading level discrepancies

sidgupta234 committed:
[MRG+1] #7466 add pandas to CI jobs (#7499)

* add pandas to CI jobs

* add pandas to CI jobs, removed errors

* commit

* commit

*  add pandas to CI jobs after review1

*  add pandas to CI jobs after review2

*  add pandas to CI jobs after review2-b

*  add pandas to CI jobs, removed extra spaces

*  add pandas to CI jobs, update test_script

*  add pandas to CI jobs, remove formatting error

*  add pandas to CI jobs, remove extranous space

amueller committed:
minor doc fixes (#7514)

fix scoring function table

minor fixes to whatsnew

ixjlyons committed:
DOC: Tidy up mds module documentation. (#7453)

* DOC: Tidy up mds module documentation.

* Add missing defaults.

jnothman committed:
DOC release date in what's new (#7518)

rickiepark committed:
add feature_name to diabetes dataset (#4477)

iamprakashom committed:
[MRG+1] minor link fix for SAG paper. (#7523)

* Link to SAG paper updated

link to SAG paper in `References` section updated (#7512).

* Link to SAG paper updated

dead link of SAG paper in Logistic Regression user guide updated.
resolves #7512

* SAG paper cited in docstring

SAG paper cited in `Reference` section of docstring(#7512).

raghavrv committed:
MNT Tags built by sublime ctags may be of form (.)tags* (#7494)

nelson-liu committed:
[MRG+1] correct condition in decision tree construction (#7441)

* remove unused condition in decision tree construction

* edit is_leaf condition for min_weight_leaf

* edit ordering of statements

* remove extra parens and add whats new

jnothman committed:
[MRG] CI build web site from branches (#7498)

* CI build web site from branches

Fixes #7495

* CI Try one regexp instead of list

kmike committed:
[MRG+1] micro-optimize HashingVectorizer and FeatureHasher (#7470)

* micro-optimize HashingVectorizer and FeatureHasher

* fix backwards compatibility for Cython < 0.20

jnothman committed:
DOC clearer what's new for min weight leaf optimisation

aesuli committed:
[MRG+1] Added 'average' option to passive aggressive classifier/regressor. (#4939)

pieteradejong committed:
[MRG+1] minor edits to multiclass documentation (#5837)

* minor edits to multiclass documentation

* minor text formatting and language modifications

* minor word changes, sentence clarifications

* attempt to fix ci test

* restored previous paragraph

* minor grammatical correction

* minor grammar fixes

jnothman committed:
DOC paraphrase multiclass description

amueller committed:
add warning about python 2.6 being no longer supported (#7532)

rombiee committed:
[MRG + 1] Raised value error in HashingVectorizer and CountVectorizer (#7507)

* Raised value error in HashingVectorizer and CountVectorizer

* error statements fixed

* added unit tests

* added unit tests and value error in tfidvect

* fixed pep8 errors

* changed assertion type

* removed fit assert from hashingvectorizer

* changed test names

* unit tests changes and redundancies removed

charltonaustin committed:
[MRG+1] Fixing error with step parameter #7467 (#7469)

* Fixing error with step parameter #7467. Basically converting the step parameter the same way RFE does.

* Adding in an explanation to the test case.

* Moving the location of where we set the data.

* Adding in an assertion against the data.

julienaubert committed:
learning_curve takes error_score parameter and propagates it. (#7397)

raghavrv committed:
BUGFIX _store train_scores only if return_train_score is True (#7535)

rubik committed:
DOC fix typo in NearestNeighbors docstring (#7545)

carlsonp committed:
extra word in code comment (#7561)

jorahn committed:
fix typo in RandomizedSearchCV docs (#7564)

add missing ' in RandomizedSearchCV docs

anguswilliams91 committed:
DOC return values of sample() function in mixture/base.py (#7571)

amueller committed:
add numpy versions for backports / fixes

lesteve committed:
FIX flake8_diff.sh git repo unshallowing (#7575)

which fixes the issue seen in
https://github.com/scikit-learn/scikit-learn/pull/7027.

Enhancements:
* the script works on PRs targetting a fork rather than the reference
  repo
* improved logic about when to unshallow the repo
* fetching the upstream master branch into a reference rather than a
  branch (less intrusive for scripts than may run afterwards e.g. the
  ones in the after_success step)
* Better logging messages

iabhi7 committed:
Rangespan testimonial has misformatted link (#7576)

jasmainak committed:
DOC FIX: update warning about random seed with scipy (#7577)

bthirion committed:
[MRG+2]: Fix Ridge GCV with centered data (#6178)

ogrisel committed:
DOC add what's new for #6178

kathyxchen committed:
[MRG+1] Docs: refer users to the other encoders to do one hot encoding for labels. (#7315)

* refer users to the other encoders to do one hot encoding for labels.

* added to the 'see more' for labelbinarizer, multilabelbinarizer, and labelencoder' as well as an example to multilabel binarizer

* added note about y labels to the OneHotEncoder docstring

* removed example from MultiLabelBinarizer

* documentation should specify LabelBinarizer, not MultiLabelBinarizer in OHE

nelson-liu committed:
[MRG + 1] fix test_20news_vectorized (#7431)

Run test if dataset has already been downloaded rather than always skipping it

Ajackster committed:
[MRG] DOC Fixed the display of search results (#7517) (#7560)

* Fixed search not expanding page

* Changed the comment to warn users why not to leave fixed height on header

jkarno committed:
DOC add spaces before colons in docstrings (#7589)

zhengruifeng committed:
[MRG] Fix GaussianProcess batch predict in Py3k  #7329 (#7330)

* fix batch

* add test for eval_MSE

elliott2081 committed:
changed address for nose, additionally, should nose2 be considered for testing since the nose project web site suggests new projects should consider using Nose2, or py.test?

TomDLT committed:
DOC add link in the contributing doc for rebasing on master (#7541)

elliott2081 committed:
got rid of extra line

gszpak committed:
[MRG] FIX Calling fit_transform instead of transform in Pipeline's fit_predict (#7585)

* BUGFIX Calling fit_transform instead of transform in Pipeline's fit_predict (#7558)

* PEP8 fixes in test_fit_predict_on_pipeline

* Added comment explaining separate estimators for pipeline in test_fit_predict_on_pipeline

jkarno committed:
[MRG+2] DOC add info to conventions for multi-label fitting (#7519)

* DOC add info to conventions for multi-label fitting

* DOC move to multilabel section and small edits

* DOC clean multilabel examples and clean information

* DOC fix line lengths for multiclass

rickiepark committed:
add a test

souravsingh committed:
Update install instructions for Windows (#7550)

lesteve committed:
hanged address for nose (#7592)

rickiepark committed:
fix error in test code

GaelVaroquaux committed:
Merge pull request #7525 from rickiepark/master

[MRG+2] add feature_name to diabetes dataset

Commit:
[MRG + 1] [TST] (half-cosmetic) use less nose.tools import to simplify future transition to py.test (#7384)

* use less nose.tools import to simplify future transition to activly developing test suites/runners

* assert_equal -> assert_array_equal in test_feature_hasher_pairs_with_string_values

and one missed ImportError that should be replaced with AttributeError

* test for py2.6 compat with except AttributeError

* fix importing of SkipTest

* force using nose in python2.6 for now

* there was no assert_dict_equal in py2.6. but we can use assert_equal

although failed test will look a little bit ugly

* remove nose imports from doc/datasets

jakevdp committed:
BUG: handle integer inputs in LLE (fixes #6033) (#6282)

rushter committed:
Fix test suites (#6348)

nelson-liu committed:
fix an indentation issue in whatsnew (#7605)

IshankGulati committed:
removed cv_ prefix (#6252)

NelleV committed:
DOC numpydoc requires space between name and column (#7601)

This patch fixes a tiny mistake in the contirbuting guidelines of the See also
section. The space between the function name and description was missing.

chenhe95 committed:
[MRG + 1] Added some documentation for loading external datasets (Issue 3808) (#7516)

* Update tutorial.rst

* Update tutorial.rst

* Update tutorial.rst

* Update tutorial.rst

* Update index.rst

* Update index.rst

* Update tutorial.rst

* Update tutorial.rst

* Update tutorial.rst

* Update faq.rst

* Update faq.rst

* Divided in two cases (standard columnar and misc data)

I also added a preprocessing note at the end

* Update tutorial.rst

* Update faq.rst

* Update index.rst

Also added some references that were in the original FAQ and pointed the FAQ to here

* Update index.rst

Added the information from the removed part of the FAQ because I felt that the FAQ version was better than the explanation I gave.

* Update index.rst

reference to skimage and also has sklearn.preprocessing.OneHotEncoder instead of OneHotEncoder

* Update index.rst

* Update index.rst

Changed with @jnothman's feedback 
https://github.com/scikit-learn/scikit-learn/pull/7516/files/d16ac523ed404188fc1f2529ac89050d4a974e3f

* Update faq.rst

* optimized file formats added to datasets/index.rst

Note: if you manage your own numerical data it is recommended to use an optimized file format such as HDF5 to reduce data load times. Various libraries such as H5Py, PyTables and pandas provides a Python interface for reading and writing data in that format.

- From the FAQ

* faq.rst: Moved the comment in bunch section to datasets index

This comment has been moved to the datasets index in the external_datasets section:
Note: if you manage your own numerical data it is recommended to use an optimized file format such as HDF5 to reduce data load times. Various libraries such as H5Py, PyTables and pandas provides a Python interface for reading and writing data in that format.

* Update index.rst

Included all changes mentioned by @amueller and @jnothman

* Update faq.rst

* Update faq.rst

lynchkl committed:
[MRG + 2] updated iris svm example colormap so the three species can be distinguished (#7611)

* updated iris svm colormap so the three species can be distinguished

* updated colormap to viridis, which is robust to colorblindness

* changing colormap to coolwarm

kgilliam125 committed:
[MRG] Documentation describes the Need Contributor label (#7599)

amueller committed:
DOC pos_label can be string (#7607)

ivallesp committed:
TST ensure relationship between predict_proba and predict_log_proba (#7612)

nelson-liu committed:
[MRG] FIX min_samples_split and min_samples_leaf validation for float vs int (#7604)

lesteve committed:
COSMIT fix typo in docstring

Commit:
DOC fixed avoidable warning of test-doc (#7524)

ogrisel committed:
DOC fix link format in reference for BayesianMixtureModel

jakevdp committed:
Add whats new entry for #6282 (#7629)

raghavrv committed:
[MRG+1] FIX Make sure GridSearchCV and RandomizedSearchCV are pickle-able (#7594)

* FIX Subclass a new MaskedArray which allows pickling even when dype=object

* TST unpickling too

* FIX Use MaskedArray from utils.fixes rather than from numpy

* FIX imports

* Don't assign a variable

* FIX np --> numpy

* Use tostring instead of tobytes for old numpy

* COSMIT pickle-able --> picklable

* use #noqa comment to turn off flake8

* TST/ENH Check if the pickled est's predict matches with the original one's

sshleifer committed:
[MRG] FIX regression: Pipeline.fit_predict uses fit_params as in 0.17.x (#7621)

jakevdp committed:
[MRG + 1] FIX: remove checks from PyFunc distance metric (fixes #6287) (#6288)

jnothman committed:
DOC What's new for #6288

Commit:
dev-7360: Explain the meaning of prefix [MRG + 1] and [MRG + 2] in the title of the pull request. (#7636)

* File updated: doc/developers/contributing.rst

toastedcornflakes committed:
Improve robustness of LFW dataset downloader (#6112)

jnothman committed:
DOC what's new for #6497 and 0.18.1 section

ogrisel committed:
DOC better docstring for TruncatedSVD (#7651)

ogrisel committed:
MAINT make appveyor fail on old builds when PR is update (#6365)

amueller committed:
DOC Removing deprecated DPGMM that was also not rendering correctly from classes. (#7606)

nvavrova committed:
DOC Fixed missing "Next" button (#7641)

GaelVaroquaux committed:
DOC: fix the copybutton on the code blocks (#7634)

The copybutton is the ">>>" on the side of code blocks that enables to
remove the ">>>" at the begining of lines as well as the outputs.

It was broken in the docs of 0.18 due to changes in sphinx. I am
backporting the js used by the Python core docs.

In addition, I am backporting CSS to highlight this button
on hover.

Commit:
dev-7369: Make common metric tests look nicer (#7620)

* Change the tests to yield after performing the _named_check.

* Tests affected:

  -test_averaging_multiclass
  -test_averaging_multilabel
  -test_averaging_multilabel_all_zeroes
  -test_averaging_multilabel_all_ones
  -test_sample_weight_invariance

* File updated: sklearn/metrics/tests/test_common.py

nelson-liu committed:
[MRG] edit releases link from sf to pypi (#7637)

* edit releases link from sf to github

* edit link to pypi

Commit:
[MRG+1] Throw an error with explicit message if n_estimators is not an integer. (#7457)

* Throw an error with explicit message if n_estimators is not an integer.

* Testing for explicit message if n_estimators is not an integer.

* Fixed typo in test for explicit message if n_estimators is an integer.

* Added tests for np.int32 and float input.

* pep8 compliance

* fix function name

* Import numpy to test n_estimators suplied as numpy int32.

olologin committed:
[MRG] fix #6101 GradientBoosting decision_function for sparse inputs (#6116)

andrewcsmith committed:
DOC: fix return value doc for mixture.base._e_step (#7682)

yangarbiter committed:
[MRG+1] FIX unstable cumsum (#7376)

* FIX unstable cumsum in utils.random

* equal_nan = true for isclose
since numpy < 1.9 sum is as unstable as cumsum, fallback to np.cumsum

* added axis parameter to stable_cumsum

* FIX unstable sumsum in ensemble.weight_boosting and utils.stats

* FIX axis problem in stable_cumsum

* FIX unstable cumsum in mixture.gmm and mixture.dpgmm

* FIX unstable cumsum in cluster.k_means_, decomposition.pca, and manifold.locally_linear

* FIX unstable sumsum in dataset.samples_generator

* added docstring for parameter axis of stable_cumsum

* added comment for why fall back to np.cumsum when np version < 1.9

* remove unneeded stable_cumsum

* added stable_cumsum's axis testing

* FIX numpy docstring for make_sparse_spd_matrix

* change stable_cumsum from error to warning

Commit:
warning for PCA with sparse input (#7649)

jstriebel committed:
[MRG+1] CircleCI timeout extended (#7693)

* Circle CI: adding 1h timeout for build_doc.sh

* Circle CI: typo + trigger build

* CircleCI: trigger build again

FERRIA committed:
DOC Changed Contributor's Guide to Development Guide #7690 (#7691)

ljwolf committed:
switch to multinomial composition for mixture sampling

lesteve committed:
DOC mention JOBLIB_START_METHOD in crash/freeze FAQ

jnothman committed:
DOC use combined unicode chars in author names (#7706)

ljwolf committed:
add shape assertions to test

NarineK committed:
[MRG+1] Learning curve: Add an option to randomly choose indices for different training sizes (#7506)

* Chooses randomly the indices for different training sizes

* Bring back deleted line

* Rewrote the description of 'shuffle' attribute

* use random.sample instead of np.random.choice

* replace tabs with spaces

* merge to master

* Added shuffle in model-selection's learning_curve method

* Added shuffle for incremental learning + addressed Joel's comment

* Shorten long lines

* Add 2 blank spaces between test cases

* Addressed Joel's review comments

* Added 2 blank lines between methods

* Added non regression test for learning_curve with shuffle

* Fixed indentions

* Fixed space issues

* Modified test cases + small code improvements

* Fix some style issues

* Addressed Joel's comments - removed _shuffle_train_indices, more test cases and added new entry under 0.19/enhancements

* Added some modifications in whats_new.rst

amueller committed:
explain learning_curve(shuffle=True) test.

olologin committed:
FIX #6420: Cloning decision tree estimators breaks criterion objects (#7680)

Commit:
[MGR + 2] fix selectFdr bug (#7490)

lesteve committed:
Update joblib to 0.10.3 version (#7696)

lesteve committed:
Remove temporary work-around for 0.15 release

See 68280fb4 for more details.

abatula committed:
[MRG+1] Added unit test for adding classes_ property to GridSearchCV, fixes #6298 (#7661)

* Fix issue #6298
Adds a "classes_" property to BaseSearchCV

* Added test to ensure classes_ property is added to gridSearch correctly

* Fixed formatting

* Added test to ensure gridSearchCV with a regressor does not have a classes_ attribute

* Fixed whitespace issues

* Combined tests for the new GridSearchSV.classes_ property into a single test.

* Removed trailing whitespace

* Added what's new for pull request #7661

* Fixed formatting of update in what's new

amueller committed:
add missing links for users to whatsnew

JungeAlexander committed:
[MRG + 1] Move n_iter and get_params invariance tests to common estimator_checks (#7677)

* Test get_params invariance in common estimator tests

Remove test_get_params_invariance() from `test_common.py` and add
test call to _yield_all_tests() in `estimator_checks.py` to make
sure that get_params(deep=False) of a given Estimator returns a
subset of get_params(deep=True).

Compared to test_get_params_invariance(), it is NOT tested anymore
whether the given Estimator has an attribute get_params since
class BaseEstimator in `base.py` defines such an attribute
for each Estimator.

Partially addresses issue #7533
Also related to issue #4465

* Move test_transformer_n_iter() to estimator_checks.py

Remove the test test_transformer_n_iter() from tests/test_common.py
and perform the test logic in utils/estimator_checks.py instead.
Specifically, the method _yield_transformer_checks() now yields
check_transformer_n_iter() as part of the set of tests for
transformers.

test_transformer_n_iter() tests that that transformers with an
attribute max_iter, return the attribute of n_iter at least 1.

Partially addresses latter part of issue #7533

* Move test_non_transformer_estimators_n_iter() to estimator_checks.py

Remove the test_non_transformer_estimators_n_iter() from
tests/test_common.py; perform the test logic in
utils/estimator_checks.py instead.
Specifically, the method _yield_non_meta_checks() now yields
check_non_transformer_estimators_n_iter().

test_transformer_n_iter() tests that that estimators that are not
transformers with an attribute max_iter, return the attribute n_iter
of at least 1.

NOTE: The current implementation makes said test run for more
estimators than before this commit.
For some of these estimators, the test fails. This needs to be addressed
(see FIXME in line 111-115 of utils/estimator_checks.py for a potential
place to start).

Partially addresses latter part of issue #7533

* Fix check_non_transformer_estimators_n_iter calls

test_transformer_n_iter() test is now only run for
estimators where the test is applicable.

Partially addresses latter part of issue #7533

* Run check_non_transformer_estimators_n_iter on multi-class estimators

To do this, use helper method multioutput_estimator_convert_y_2d.
Also remove multi_output parameter from
check_non_transformer_estimators_n_iter since this parameter is not
used anywhere and corresponding cases should be handled by said
helper method.

Also, some pep8 line length fixes.

* Fix documentation for n_iter tests

There was some confusion between attributes and parameters.
Also rename n_iter to n_iter_

lesteve committed:
Use n_components=3 to test actual regression

n_components and n_features were equal and one was used for the other in
some places.

affanv14 committed:
[MRG+2] adding multilabel support for score_func (#7676)

* added multilabel support for score function

* added test for multilabel score function

* updated whats_new.rst

* updated whats_new.rst with working link

kgilliam125 committed:
[MRG] Added warning on keyboard interrupt during MLP fit (#7614)

lesteve committed:
[MRG+2] switch to multinomial composition for mixture sampling (#7702)

* switch to multinomial composition for mixture sampling

* add shape assertions to test

* Use n_components=3 to test actual regression

n_components and n_features were equal and one was used for the other in
some places.

amueller committed:
removed parameter that was documented as attribute (#7711)

mblondel committed:
Fix typo in OMP author name.

First author is Stéphane Mallat.

tguillemot committed:
[MRG+1] Reorder EllipticEnvelope docstring. (#7734)

* Reorder EllipticEnvelope docstring.

* Fix pep8

srivatsan-ramesh committed:
[MRG + 1] Printing the total time in cross_validation (#7640)

* print score+fit time instead of just score time when doing cross_validation

* reducing line size

* More clearer log message

antoinewdg committed:
[MRG+2] Norm inconsistency between RFE and SelectFromModel (was _LearntSelectorMixin) #2121 (#6181)

* Norm inconsistency between RFE and SelectFromModel (was _LearntSelectorMixin) #2121

* safe_pwr utility

* Norm fix

* Removed safe_pwr

* 1D arrays support for norm fix

* Test case for 2d coef in SelectFromModel

* Fix numpy version requirement for norm fix

* Implement fixes suggested by @jnothman

* Add numpy version requiring the fix.

chkoar committed:
Address #7733 - MultiTaskElasticNet user guide links to MultiTaskLasso (#7741)

bburns committed:
r2_score - add more doctest examples (#7727)

Add simple examples showing perfect score, expected value score, and negative score.

kgilliam125 committed:
[MRG + 1] Clarified error msg in plot_partial_dependence (#7673)

* Clarified error msg in plot_partial_dependence

* Changed err msg for feature[i] out of range. Updated docs.

* Error message shows invalid value.

GaelVaroquaux committed:
BF: avoid importing from inside joblib (#7731)

Debian replaces externals.joblib by an import to joblib, because they
hate duplication. Hence importing from inside joblib doesn't work, and
the line that I removed greats a bug.

See
http://lists.alioth.debian.org/pipermail/neurodebian-users/2016-October/001093.html

erikcs committed:
[MRG + 1] ElasticNetCV: raise ValueError if l1_ratio=0 (#7591)

Raise ValueError if l1_ratio=0 in ElasticNetCV and alphas=None

Deborah-Digges committed:
[MRG+2] logit -> logistic in plot_logistic.py and minor visual improvements to the plot. (#7730)

* Small correction of logit->logistic

The function `1 / (1 + np.exp(-x))` is the logistic function. 
The logit function is the inverse of the logistic function : `log(x/(1-x))`

* Add axes ticks, legend and colors to the plot

* Fix flake8 errors

* Rename legend labels as per review

* Fix pep8 error

JPFrancoia committed:
[MRG] Correcting length of explained_variance_ratio_, eigen solver (#7632)

jnothman committed:
[MRG] DOC :issue: role to simplify what's news (#7657)

amueller committed:
[MRG+1] replaced some assert_true(np.allclose(x, y)) with assert_almost_equal (#7742)

* replaced some assert_true(np.allclose(x, y)) with assert_almost_equal for better error messages.

also some pep8.

* typo fixes

kmike committed:
DOC use target_names over named categories in 20newsgroups example (#7423)

raghavrv committed:
DOC Correct linking of TruncatedSVD (#7749)

polmauri committed:
[MRG + 1] FIX raise an error message when n_groups > number of groups (#7681) (#7683)

* FIX raise an error message when n_groups > actual number of groups (#7681)

This change addresses issue #7681:
- Raise ValueError when n_groups > actual number of unique groups in LeaveOneGroupOut and LeavePGroupsOut.
- Add unit test.

* Make requested changes

- Check error message with `assert_raise_message`
- Pass parameters to `assert_raise_message` instead of defining functions

* Update condition and exception message

ngoix committed:
[MRG+2] LOF algorithm (Anomaly Detection) (#5279)

* LOF algorithm

add tests and example

fix DepreciationWarning by reshape(1,-1) one-sample data

LOF with inheritance

lof and lof2 return same score

fix bugs

fix bugs

optimized and cosmit

rm lof2

cosmit

rm MixinLOF + fit_predict

fix travis - optimize pairwise_distance like in KNeighborsMixin.kneighbors

add comparison example + doc

LOF -> LocalOutlierFactor
cosmit

change LOF API:
-fit(X).predict() and fit(X).decision_function() do prediction on X without
 considering samples as their own neighbors (ie without considering X as a
 new dataset as does fit(X).predict(X))
-rm fit_predict() method
-add a contamination parameter st predict returns a binary value like other
 anomaly detection algos

cosmit

doc + debug example

correction doc

pass on doc + examples

pep8 + fix warnings

first attempt at fixing API issues

minor changes

takes into account tguillemot advice

-remove pairwise_distance calculation as to heavy in memory
-add benchmarks

cosmit

minor changes + deals with duplicates

fix depreciation warnings

* factorize the two for loops

* take into account @albertthomas88 review and cosmit

* fix doc

* alex review + rebase

* make predict private add outlier_factor_ attribute and update tests

* make fit_predict take y argument

* fix benchmarks file

* update examples

* make decision_function public (rm X=None default)

* fix travis

* take into account tguillemot review + remove useless k_distance function

* fix broken links :meth:`kneighbors`

* cosmit

* whatsnew

* amueller review + remove _local_outlier_factor method

* add n_neighbors_ parameter the effective nb neighbors we use

* make decision_function private and negative_outlier_factor attribute

amueller commented:
@alexsavio do you want to finish this or have someone else take it up?

maniteja123 committed:
DOC: Provide link to LDA and NMF in the example tutorial closes #5876 (#5984)

asanakoy committed:
[MRG + 1] fix bug with negative values in cosine_distances (#7732)

* fix bug with negative values in cosine_distances
clip distances to [0, 2]
set distances between vectors and themselves to 0

* add test

* add test on big random matrix

* use np.diag_indices_from instead of slicing

alexsavio committed:
Merge remote-tracking branch 'refs/remotes/alex/issue4348v2' into issue4348v2

alexsavio committed:
add check_random_state_shuffle and test

Commit:
[MRG+2] DOC correct docstring for sample_gaussian (#6957)

* DOC correct docstring for sample_gaussian

Docstring did not match function behaviour. Caused some trouble trying to implement a compatible version for a different distribution.

* MAINT Use _sample_gaussian internally

This avoids triggering the deprecation warning. sample_gaussian is just a thin
wrapper.

* Fix _sample_gaussian test path

_sample_gaussian is not publicly exported in mixture, so use gmm module.

amueller commented:
the test is failing and there seems to be an issue with the review. Make sure you update your master from upstream before rebaseing.

amueller commented:
see docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#rebasing-on-master

alexsavio committed:
fix: replace tab to spaces in cross_validation.py

alexsavio commented:
I did a rebase from upstream as it is explained in the docs. The error was that the filemerger (meld) I used inserted a tab instead of 4 spaces for an indentation. The sad thing is that Python2 accepted it.

varunshenoy committed:
Added citation for sklearn (#7759)

amueller commented:
Hm... I mean we can hope that with ``squash and merge`` this goes away....

amueller commented:
na there is 700 changed files. Not sure, I never have issues with meld.
maybe do
```
git checkout master
git pull upstream master
git checkout issue4348v2
git reset master --mixed
```
and then add and commit the actual changes.

perimosocordiae committed:
[MRG+1] BUG: MultiLabelBinarizer.fit_transform sometimes returns an invalid CSR matrix (#7750)

* BUG: MultiLabelBinarizer makes invalid CSR matrix

See https://github.com/scipy/scipy/issues/6719 for context.

The gist is that the `inverse` array may have a different dtype than `yt.indices`, which causes trouble down the line because, in those cases, `yt.indices` and `yt.indptr` have different dtypes.

Alternately, we could insert `yt.check_format(full_check=False)` after modifying the sparse matrix members.

* Fixing for old numpy

Older versions don't support kwargs for `astype`

* Adding tests

* line-wrapping

* adding comment to tests

[ci skip]

* added rationale comment

[ci skip]

alexsavio committed:
add check random_state vs. shuffle

alexsavio committed:
Merge remote-tracking branch 'refs/remotes/alex/issue4348v2' into issue4348v2

alexsavio commented:
@amueller, I did what you asked, sort of. If this is not what you expected, I offer to do a new PR.

amueller commented:
Sort of?
Please open a new PR then. I don't want to change all the files in sklearn with this PR.

alexsavio commented:
Yes, your suggestion brought other issues in my repository when I did the reset.
